src/mock-doc/node.ts

```diff
@@ -261,9 +266,11 @@ export class MockElement extends MockNode {
     }
   }
 
-  get attributes() {
+  get attributes(): MockAttributeMap {
     if (this.__attributeMap == null) {
-      this.__attributeMap = createAttributeProxy(false);
+      const attrMap = createAttributeProxy(false);
+      this.__attributeMap = attrMap;
+      return attrMap;
```

Comment:
Non blocking - we could probably simplify this fn by not early returning:
```suggestion
      this.__attributeMap = createAttributeProxy(false);
```
by directly assigning to `__attributeMap` and letting the return statement outside of the `if` take care of the rest

Reply:
Disregard, I missed the comments inline somehow 🤔